android_lib/src/main/java/io/sentry/godotplugin/SentryAndroidGodotPlugin.kt

```diff
@@ -0,0 +1,394 @@
+            Log.e(TAG, "Internal Error -- SentryException not found: $exceptionHandle")
+        }
+        return exception
+    }
+
+    private fun registerEvent(event: SentryEvent): Int {
+        val eventsMap = eventsByHandle.get() ?: run {
+            Log.e(TAG, "Internal Error -- eventsByHandle is null")
+            return 0
+        }
+
+        var handle = Random.nextInt()
+        while (eventsMap.containsKey(handle)) {
+            handle = Random.nextInt()
+        }
+
+        eventsMap[handle] = event
+        return handle
+    }
+
+    private fun registerException(exception: SentryException): Int {
+        val exceptionsMap = exceptionsByHandle.get() ?: run {
+            Log.e(TAG, "Internal Error -- exceptionsByHandle is null")
+            return 0
+        }
+
+        var handle = Random.nextInt()
+        while (exceptionsMap.containsKey(handle)) {
+            handle = Random.nextInt()
+        }
+
+        exceptionsMap[handle] = exception
+        return handle
+    }
+
+    override fun getPluginName(): String {
+        return "SentryAndroidGodotPlugin"
+    }
+
+    @UsedByGodot
+    fun initialize(
+        beforeSendHandlerId: Long,
+        dsn: String,
+        debug: Boolean,
+        release: String,
+        dist: String,
+        environment: String,
+        sampleRate: Float,
+        maxBreadcrumbs: Int,
+    ) {
+        Log.v(TAG, "Initializing Sentry Android")
+        SentryAndroid.init(godot.getActivity()!!.applicationContext) { options ->
+            options.dsn = dsn.ifEmpty { null }
+            options.isDebug = debug
+            options.release = release.ifEmpty { null }
+            options.dist = dist.ifEmpty { null }
+            options.environment = environment.ifEmpty { null }
+            options.sampleRate = sampleRate.toDouble()
+            options.maxBreadcrumbs = maxBreadcrumbs
+            options.sdkVersion?.name = "sentry.java.android.godot"
+            options.nativeSdkName = "sentry.native.android.godot"
+            options.beforeSend =
+                SentryOptions.BeforeSendCallback { event: SentryEvent, hint: Hint ->
+                    Log.v(TAG, "beforeSend: ${event.eventId} isCrashed: ${event.isCrashed}")
+                    val handle: Int = registerEvent(event)
+                    Callable.call(beforeSendHandlerId, "before_send", handle)
+                    eventsByHandle.get()?.remove(handle) // Returns the event or null if it was discarded.
+                }
+            }
+    }
+
+    @UsedByGodot
+    fun addGlobalAttachment(path: String) {
+        val attachment = Attachment(path)
+        Sentry.getGlobalScope().addAttachment(attachment)
+    }
+
+    @UsedByGodot
+    fun setContext(key: String, value: Dictionary) {
+        Sentry.getGlobalScope().setContexts(key, value)
+    }
+
+    @UsedByGodot
+    fun removeContext(key: String) {
+        // Q: Godot SDK doesn't expose scopes yet, should we set it on the global scope?
```

Comment:
yes, just global since Godot is a desktop/mobile app and we default to global mode in these scenarios.

thread-local is the default on server apps